spec/models/ab_experiment/goal_conversion_handler_spec.rb

```diff
@@ -0,0 +1,151 @@
+
+      before do
+        field_test(AbExperiment::CURRENT_FEED_STRATEGY_EXPERIMENT, participant: user)
+      end
+
+      it "records a conversion" do
+        handler.call
+        expect(FieldTest::Event.last.field_test_membership.participant_id).to eq(user.id.to_s)
+        expect(FieldTest::Event.last.name).to eq(goal)
+      end
+
+      it "records user_creates_comment_on_at_least_four_different_days_within_a_week field test conversion" do
+        7.times do |n|
+          create(:comment, user_id: user.id, created_at: n.days.ago)
+        end
+        handler.call
+        expect(FieldTest::Event.last.field_test_membership.participant_id).to eq(user.id.to_s)
+        expect(FieldTest::Event.last.name)
+          .to eq("user_creates_comment_on_at_least_four_different_days_within_a_week")
+      end
+    end
+
+    context "with user who is part of field test and user_creates_pageview goal" do
+      let(:goal) { described_class::USER_CREATES_PAGEVIEW_GOAL }
+
+      before do
+        field_test(AbExperiment::CURRENT_FEED_STRATEGY_EXPERIMENT, participant: user)
+      end
+
+      it "records user_views_pages_on_at_least_four_different_days_within_a_week field test conversion" do
+        7.times do |n|
+          create(:page_view, user_id: user.id, created_at: n.days.ago)
+        end
+        handler.call
+        expect(FieldTest::Event.last.field_test_membership.participant_id).to eq(user.id.to_s)
+        expect(FieldTest::Event.pluck(:name))
+          .to include("user_views_pages_on_at_least_four_different_days_within_a_week")
+      end
+
+      it "records user_views_pages_on_at_least_nine_different_days_within_two_weeks field test conversionn" do
+        10.times do |n|
+          create(:page_view, user_id: user.id, created_at: n.days.ago)
+        end
+        handler.call
+        expect(FieldTest::Event.last.field_test_membership.participant_id).to eq(user.id.to_s)
+        expect(FieldTest::Event.pluck(:name))
+          .to include("user_views_pages_on_at_least_nine_different_days_within_two_weeks")
+      end
+
+      it "records user_views_pages_on_at_least_twelve_different_hours_within_five_days field test conversion" do
+        15.times do |n|
+          create(:page_view, user_id: user.id, created_at: n.hours.ago)
+        end
+
+        handler.call
+        expect(FieldTest::Event.last.field_test_membership.participant_id).to eq(user.id.to_s)
+        expect(FieldTest::Event.pluck(:name))
+          .to include("user_views_pages_on_at_least_twelve_different_hours_within_five_days")
+      end
+
+      it "does not record field test conversion if not qualifying" do
+        2.times do |n|
+          create(:page_view, user_id: user.id, created_at: n.days.ago)
+        end
+        handler.call
+        expect(FieldTest::Event.all.size).to be(0)
+      end
+
+      it "records user_views_pages_on_at_least_four_different_hours_within_a_day field test conversionn" do
```

Comment:
Quick, non-blocking suggestion (same as above!): let's aggregate the failures here since we have multiple expectations:
```suggestion
      it "records user_views_pages_on_at_least_four_different_hours_within_a_day field test conversionn", :aggregate_failures do
```